### PR TITLE
Fixed #26373 -- Fixed reverse lookup crash with a ForeignKey to_field in a subquery.

### PR DESCRIPTION
**Pull request by Moncader**
This fixes a regression from #6117. 

I haven't looked too deep in to the details of this. So forgive me if the fix is a bit quick and hacky (if it's fine then great!).

Any feedback is welcome.

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: tests/queries/tests.py [hunk 1]
```diff
@@ -2427,6 +2427,14 @@ def test_in_subquery(self):
             set(Eaten.objects.filter(food__in=Food.objects.filter(name='apple').values('eaten__meal'))),
             set()
         )
+        self.assertEqual(
+            set(Food.objects.filter(eaten__in=Eaten.objects.filter(meal='lunch'))),
+            {apple}
+        )
+        self.assertEqual(
+            set(Food.objects.filter(eaten__in=Eaten.objects.filter(meal='lunch').values('food__name'))),
+            set()
```
Comment (charettes) [inline tests/queries/tests.py]: Shouldn't this be equal to `{apple}`?

**Comment by charettes** [comment]
Hi @Moncader thanks for the patch!

As this is not considered a trivial fix this will require the creation of a [Trac ticket](https://code.djangoproject.com/newticket).

Please details if any exception occur or which query result changed.

**Comment by Moncader** [comment]
Sure thing. Will do first thing tomorrow.

File: tests/queries/tests.py (hunk 1 above)
Comment (Moncader) [inline tests/queries/tests.py]: Yes. It should. However so should the assert above this from the previous version as far as my knowledge of values works (IE. It shouldn't change the query).

As I believe that is yet a separate issue, I've left that test there and just copied it down for the exact reverse test as above.

I can remove it if you would prefer though. (Technically it is incorrect.)

**Comment by Moncader** [comment]
https://code.djangoproject.com/ticket/26373
Created a trac ticket.

**Comment by timgraham** [comment]
Did you see the test failure on PostgreSQL?

**Comment by Moncader** [comment]
Now I have yes. Not sure how that change would cause a backend specific error but I'll look in to it when I have a chance on Tuesday (long weekend here).

**Comment by Moncader** [comment]
The postgresql error was due to the last test case. That test case itself was incorrect and was producing odd SQL (by setting values() yourself in a subquery is it ok for django to use that field instead of the proper field for the where in condition??)

The second assert in that test case does the same thing, but the data type is the same and postgresql lets it pass... though it's a fluke.

**Comment by Moncader** [comment]
Ugh. Looks like github didn't understand my git commit --amend... There is no conflict, and this would be resolved by re-creating this PR. You should also be able to manually merge without conflict. Would you like me to re-create it or can you manually merge?

**Comment by timgraham** [comment]
Not a problem, the conflict was just in the release notes. Merged in 4c1c93032f4a015cbb4b33958603d18ac43515b4, thanks!

**Pull request closed without merging**